packages/lime-system/files/usr/lib/lua/lime/utils.lua

```diff
@@ -469,16 +485,20 @@ function utils.get_bathost(mac, outgoing_iface)
 		local line = bathosts:read("*l")
 		while line and hostname == nil do
 			if line:find("^" .. mac) then
-				hostname = line:match(mac .. "(.*)_wlan"):gsub("_", "-"):gsub("%s+", "")
-				iface = line:match("wlan.*"):gsub("_", "-"):gsub("%s+", "")
+				hostname, iface = utils.bathost_deserialize(line)
 			end
 			line = bathosts:read("*l")
 		end
 	end
-	if hostname == nil then
-		local ipv6ll = utils.mac2ipv6linklocal(mac) .. "%" .. outgoing_iface
-		hostname = utils.http_client_get("http://[" .. ipv6ll .. "]/cgi-bin/hostname", 10):gsub("\n", "")
+	if hostname == nil and outgoing_iface then
+		iface = outgoing_iface
```

Comment:
We cannot say that the neighbor iface will be the same as our outgoing_iface
For instance in the case that wlan1_mesh of node A is associated with wlan2_mesh of node B.

Before, iface was not in the returned table. Do you have a better idea? :)

Reply:
Yes indeed, I was confused on what was the data in the bat-hosts.